### DISCUSSION TITLE
Make Date* controls use 2-digits for day and month (and other improvements)

### DISCUSSION THREAD

**Discussion by tomsontom** [adobe/react-spectrum#3436] [Ideas]
Hi,

Comparing the Date-Input types with https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date - it would be nice if one could control that for day and month 2-digits are used (this would reduce the white-gaps). 

Another nice to have would be if a monospaced font could be used (the native control uses also monospace if not mistaken) but I understand that this probably won't match the Spectrum styles.

Current:
<img width="546" alt="Bildschirmfoto 2022-08-24 um 01 35 37" src="https://user-images.githubusercontent.com/52631/186284341-72ab2340-06b9-4a37-ba00-b26a06bd4b9a.png">

2-digits:
<img width="546" alt="Bildschirmfoto 2022-08-24 um 01 36 49" src="https://user-images.githubusercontent.com/52631/186284499-44141ceb-8518-4707-873f-b5c9f32c9ad4.png">

2-digits + monospace (used Source Code Pro):
<img width="546" alt="Bildschirmfoto 2022-08-24 um 01 34 00" src="https://user-images.githubusercontent.com/52631/186284536-ff6da652-0095-4772-ad5f-785f176f1217.png">

**Comment by devongovett** [comment]
Zero prefixing is locale specific. It is expected in some regions and not others, so it happens automatically depending on that.

**Reply by tomsontom** [reply]
Well I think this 1 or 2-digit thing is true if you display a date in a label but not if you edit a date. 

I can once more only point to the native input who also uses 2-digits.

**Reply by tomsontom** [reply]
Oh and my locale is German/Austrian

**Reply by devongovett** [reply]
AFAICT chromes native date picker isn't localized at all though, so I'm not sure it's a good example to follow.

**Reply by tomsontom** [reply]
well FF is

<img width="561" alt="Bildschirmfoto 2022-08-24 um 22 55 45" src="https://user-images.githubusercontent.com/52631/186522109-d8d4b71d-206c-44e1-9b49-f112a440cfa1.png">

note the german terms for the day names and that the week starts with monday? And that the order is dd.mm.yyyy (can also be seen by the format if empty)
<img width="561" alt="Bildschirmfoto 2022-08-24 um 22 58 30" src="https://user-images.githubusercontent.com/52631/186522386-88cc05a4-ebca-437a-8ae9-539313c5e7cd.png">

and Chrome is too

<img width="725" alt="Bildschirmfoto 2022-08-24 um 23 00 16" src="https://user-images.githubusercontent.com/52631/186522605-e724d7e5-6d99-486b-8a1e-39f5b682bf32.png">

<img width="787" alt="Bildschirmfoto 2022-08-24 um 23 02 17" src="https://user-images.githubusercontent.com/52631/186522946-865eaad5-ea64-4fe7-95d8-6012e7aa12ae.png">

**Comment by tomsontom** [comment]
@devongovett Not sure you saw my latest comments native chrome and Firefox (both locale aware) disagree with your statement on 0 prefixes.
